File: api/src/main/java/io/github/wulkanowy/api/generic/Subject.java
package io.github.wulkanowy.api.generic;

public class Subject {

    private int id = -1;

    private String name = "";

    public int getId() {
        return id;
    }

    public Subject setId(int id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Subject setName(String name) {
        this.name = name;
        return this;
    }
}

File: app/src/main/java/io/github/wulkanowy/data/db/entities/Subject.kt
package io.github.wulkanowy.data.db.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "Subjects")
data class Subject(

    @ColumnInfo(name = "student_id")
    val studentId: Int,

    @ColumnInfo(name = "diary_id")
    val diaryId: Int,

    @ColumnInfo(name = "real_id")
    val realId: Int,

    val name: String
) : Serializable {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
}
